tests/functional/utils/parametrization.py

```diff
@@ -0,0 +1,29 @@
+import os
+import socket
+import random
+from datetime import datetime
+
+RAND_SEED = random.seed(datetime.now())
+
+
+def next_free_port(min_port=1024, max_port=65535):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    while min_port <= max_port:
+        try:
+            sock.bind(('', min_port))
+            sock.close()
+            return min_port
+        except OSError:
+            min_port += 1
+    raise IOError('no free ports')
+
+
+def get_ports_for_fixture():
```

Comment:
I think it might be sensitive to hazards when triggered in parallel. There is a risk that different test will get the same port assigned. Maybe more reliable would be assigning different last 2 digits for each test and 2 digits for each image target?
This verification could be handy to avoid conflicts with permanently occupied ports on the host running the tests.

Reply:
This is why I made some randomization with seed but I agree that this not fully prevents us from potential doubled assignment. Your proposition sounds nice.